source/extensions/tracers/zipkin/zipkin_tracer_impl.cc

```diff
@@ -53,30 +53,32 @@ Tracing::SpanPtr ZipkinSpan::spawnChild(const Tracing::Config& config, const std
 Driver::TlsTracer::TlsTracer(TracerPtr&& tracer, Driver& driver)
     : tracer_(std::move(tracer)), driver_(driver) {}
 
-Driver::Driver(const Json::Object& config, Upstream::ClusterManager& cluster_manager,
-               Stats::Store& stats, ThreadLocal::SlotAllocator& tls, Runtime::Loader& runtime,
+Driver::Driver(const envoy::config::trace::v2::ZipkinConfig& zipkin_config,
+               Upstream::ClusterManager& cluster_manager, Stats::Store& stats,
+               ThreadLocal::SlotAllocator& tls, Runtime::Loader& runtime,
                const LocalInfo::LocalInfo& local_info, Runtime::RandomGenerator& random_generator,
                TimeSource& time_source)
     : cm_(cluster_manager), tracer_stats_{ZIPKIN_TRACER_STATS(
                                 POOL_COUNTER_PREFIX(stats, "tracing.zipkin."))},
       tls_(tls.allocateSlot()), runtime_(runtime), local_info_(local_info),
       time_source_(time_source) {
 
-  Upstream::ThreadLocalCluster* cluster = cm_.get(config.getString("collector_cluster"));
+  Upstream::ThreadLocalCluster* cluster = cm_.get(zipkin_config.collector_cluster());
   if (!cluster) {
     throw EnvoyException(fmt::format("{} collector cluster is not defined on cluster manager level",
-                                     config.getString("collector_cluster")));
+                                     zipkin_config.collector_cluster()));
   }
   cluster_ = cluster->info();
 
-  const std::string collector_endpoint =
-      config.getString("collector_endpoint", ZipkinCoreConstants::get().DEFAULT_COLLECTOR_ENDPOINT);
+  std::string collector_endpoint = ZipkinCoreConstants::get().DEFAULT_COLLECTOR_ENDPOINT;
+  if (zipkin_config.collector_endpoint().size() > 0) {
+    collector_endpoint = zipkin_config.collector_endpoint();
+  }
 
-  const bool trace_id_128bit =
-      config.getBoolean("trace_id_128bit", ZipkinCoreConstants::get().DEFAULT_TRACE_ID_128BIT);
+  const bool trace_id_128bit = zipkin_config.trace_id_128bit();
```

Comment:
Wondering whether this field should be changed to BoolValue wrapper? The default is currently false, but in the future we may want to change that to true as 128bit ids are becoming more common.

Reply:
It might not worth, changing bool to BoolValue is a backward incompatible change so you'll need to follow deprecation process, changing default to true is another breaking change I guess?

Reply:
Agree - so will leave for now and wait for it to be raised as an issue.